tls/s2n_early_data.c

```diff
@@ -167,6 +167,18 @@ S2N_RESULT s2n_early_data_get_server_max_size(struct s2n_connection *conn, uint3
     return S2N_RESULT_OK;
 }
 
+int s2n_connection_set_server_early_data_context(struct s2n_connection *conn, const uint8_t *context, uint16_t context_size)
+{
+    POSIX_ENSURE_REF(conn);
+    if (context_size > 0) {
+        POSIX_ENSURE_REF(context);
+    }
```

Comment:
Shouldn't we error/exit when `context size  <=  0 ` here? I think we should replace this check with: 

```
POSIX_ENSURE_GT(context_size, 0);
POSIX_ENSURE_REF(context); 
```

Reply:
A uint16_t is unsigned, and a zero-length context is a valid use case. Why should we error on it?

Reply:
If it is zero-length we are not setting any context.